### PR TITLE
Update quickstart to Core 2

### PR DESCRIPTION
**Pull request by octoper**
This PR updates the quickstart to use [Core 2](https://clerk.com/changelog/2024-04-19)

### CODE REVIEW

File: src/middleware.ts [hunk 1]
```diff
@@ -1,11 +1,16 @@
-import { authMiddleware } from "@clerk/nextjs";
+import { clerkMiddleware, createRouteMatcher } from "@clerk/nextjs/server";
+
+const isProtectedRoute = createRouteMatcher([
+  '/(.*)',
```
Comment (desiprisg) [inline src/middleware.ts:4]: Does this mean that the sign in page is protected?

Comment (octoper) [reply src/middleware.ts:4]: There is no sign in page in the quickstart, it should just redirect to AP

Comment (octoper) [reply src/middleware.ts:4]: @dimkl  I just added `createRouteMatcher` so if any user that clones the repo can better understand how to protect routes later on, let me know if that not feels right

**Comment by royanger** [comment]
Hey team. There was a branch with this work already done that just needed package updates. Let me compare the two and comment on this PR as needed.

**Comment by octoper** [comment]
I didn't know that, thank you @royanger !

**Review by royanger — APPROVED** [review]
Looks good!

**Pull request merged**